lms/views/api/exceptions.py

```diff
@@ -238,7 +238,28 @@ def __json__(self, _request):
         JSON-serialize for the response body. See:
         https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/renderers.html#using-a-custom-json-method
         """
-        return {key: value for key, value in asdict(self).items() if value is not None}
+        body = {key: value for key, value in asdict(self).items() if value is not None}
+
+        # If the exception is refreshable add refresh API info for the frontend.
+        # This tells the frontend to try making a refresh request then
+        # re-trying the original request.
+        if request.feature("frontend_refresh") and getattr(
+            request.exception, "refreshable", False
+        ):
+            oauth2_token_service = request.find_service(name="oauth2_token")
+
+            try:
+                oauth2_token_service.get()
```

Comment:
TODO: Add `@lru_cache` to `OAuth2TokenService.get()` so we're not doing a duplicate DB query here (`get()` will already have been called earlier in the request)

Reply:
Separate PR for that?

Reply:
Yep: https://github.com/hypothesis/lms/pull/3714